### DISCUSSION TITLE
Airbus Oil Storage Detection model

### DISCUSSION THREAD

**Discussion by AlainFELER** [PUTvision/qgis-plugin-deepness#194] [Q&A]
I am trying many times the oil_storage model to understand if I could get something similar. 
I tried only  small places (1 to 50 km²) where there are some, and it worked very fine at big scale (1:5000 max), finding even some storages that I had not seen, but not very well at smaller scales.
A picture :
![deepness_oil](https://github.com/user-attachments/assets/9f5af223-1a92-4610-ac8e-906947078c8e)
I have some questions ... 
- in fact it seems to detect round things with a dark surrounding : a shadow or something contrasted, not specifically fuel tanks : pools, trafic circle, waste water treatment plant, isolated round trees... To understand what does the model one can only look at the training images and get some "feeling" about that ? 
- why does it identify some object and not the next one, very similar (cf my image)
- in GIS size matters : is there a way to specify that the things I look for have possibles size from 6 to 20 meters but not 3 or 50 ?
- is it possible to add the concept of mask like in remote sensing (for instance a cloud mask) ? It would be quite useful : for instance with the car model we might be looking only for the cars on the roads. Of course we could filter out the results in place without interest, but for a big search it would be faster if we could filter the tiles of the raster cover before applying the model to them
- sometimes the plugin makes QGIS crash
- I used Google images as input layer because it is said in the ZOO that the model is made upon that. If I use an orthophoto should I put the precision to that of the ortho (20 cm/px), or keep the default to 100 cm/pixel because the model need that : I don't know where to find the answer. Will we get better results with a more precise image, or will it be only more work to make the model which is a kind of compact resume of the abstract thing I look for, so more details will be lost anyway ?
- to examine the model_output it would be nice if the layer had an id (for instance the row_number) as value, because it would be easier to browse the results. I can do that myself, but for many tries with different value of confidence, iou threshold and places, it is boring to do it each time.

Thanks in advance for any answer.
AF

**Comment by przemyslaw-aszkowski** [comment]
Hi,
* "To understand what does the model one can only look at the training images and get some "feeling" about that " - generally yes, the model is as good as the data used to train it. If you are interested in improving this model, you can extend the dataset and retrain the model. Especially with "negative" examples, so that the model will now what to ignore. There are also Explainable AI technologies, but you would need some programming background to use it.
* "why does it identify some object and not the next one, very similar (cf my image)" - probably the model gives different probabilities for those objects, you can try decreasing/increasing the confidence threshold in deepness. This is an inherent feature of YOLO models.
*  is there a way to specify that the things I look for have possibles size from 6 to 20 meters but not 3 or 50 - you would need to utilize this knowledge during the training process of the model. Once the model is trained you can only do postprocessing
* "sometimes the plugin makes QGIS crash" - sounds probable, please capture logs from the crash if possible and share withus
* "or keep the default to 100 cm/pixel because the model need that" - definitively keep the same resolution cm/px. To actually utilise a better resolution you would need to train the model on a higher resolution in the first place
* "layer had an id" - interesting idea. Do you think they should be something like UUID or just an increasing number? We may implement it once we have some time :)

Cheers
Przemek

**Comment by AlainFELER** [comment]
Thank you for your answers.
- "to improve the model you can extend the dataset..." --> in fact I would like to do that, but I have no idea how.  

> roboflow then ultralytics then export to onnx-file then some tweaks
says fredrikbacke92... well, i'll try that 

- "identify some object but not the next one ; try other values of confidence --> I tried 0,1 thru 1,0 by 0,1. The only value recognising the 2 blacks circles was 0,1, which is nice but in fact those black things are **not** oil tanks, so in fact the model should not recognize them at all. I must put confidence at 0,9 to eliminate them, but then it omits also some real oil tanks...
My test place is at 149183,6835810 in EPSG:2154, zoom 1:2000 

- size : "utilize the size during the training of the model" or during postprocessing : ok
- crash : send the logs : ok
